modules/s3-bucket/main.tf

```diff
@@ -1,53 +1,83 @@
 locals {
   enabled = module.this.enabled
 
+  aws_partition = data.aws_partition.current.partition
+
   custom_policy_account_arns = [
     for acct in var.custom_policy_account_names :
-    format("arn:%s:iam::%s:root", data.aws_partition.current.partition, module.account_map.outputs.full_account_map[acct])
+    format("arn:%s:iam::%s:root", local.aws_partition, module.account_map.outputs.full_account_map[acct])
   ]
 
-  bucket_policy = var.custom_policy_enabled ? data.aws_iam_policy_document.custom_policy[0].json : var.policy
+  bucket_policy = var.custom_policy_enabled ? data.aws_iam_policy_document.custom_policy[0].json : data.template_file.bucket_policy.rendered
 }
 
 data "aws_partition" "current" {}
 
+data "template_file" "bucket_policy" {
+  template = module.bucket_policy.json
+
+  vars = {
+    id = module.this.id
+  }
+}
+
+module "bucket_policy" {
+  source  = "cloudposse/iam-policy/aws"
+  version = "0.3.0"
+
+  iam_policy_statements = var.source_policy_documents
+
+  context = module.introspection.context
+}
+
 module "s3_bucket" {
```

Comment:
@milldr did you mean to include the S3 bucket changes into this PR?

Reply:
I did not! Updating this PR today and will hopefully get this pushed through